### ISSUE TITLE
Changes to shared owncloud addressbook are not synchronized

### ISSUE THREAD

**Issue by vgencda** [jobisoft/DAV-4-TbSync#29]
Greetings,

if you need more information let me know. Thanks in advance.

Versions:
* owncloud 10.0.9.5
* Thunderbird 60.3.0
* TbSync 0.7.16
* sabre/dav (CalDAV & CardDAV) provider for TbSync 0.8.29

Steps to reproduce:

1. Create an owncloud addressbook for the admin user.
2. Share it with a group of users (non-admins).
3. Let a member of this user group subscribe to the addressbook using TbSync.
4. Change an entry, delete one or add a new one using the Thunderbird addressbook dialog.
5. Trigger synchronization to owncloud server using TbSync.

Result:
* TbSync reports clean sychronization state.
* Changes persist in Thunderbird addressbook even when synchronzing more times afterwards.
* No changes on the owncloud server side an thus for any user or client apart from the Thunderbird instance used to make the changes.

**Comment by jobisoft** [comment]
Can I have a debug log of such a failed up-sync? I need to see the server responses... 
https://github.com/jobisoft/TbSync/wiki/Found-a-bug%3F

Also, any chance I can have a test account on your server, where you setup an environment, where I can reproduce this?

**Comment by vgencda** [comment]
Unfortunately a test account is not feasible. But I will send you a log and also think I have got an idea regarding the cause of the bug. TbSync seems to try syncing with 

owncloud.domain/remote.php/carddav/addressbooks/username/contacts_shared_by_adminusername/ 

whereas the correct URL would be 

owncloud.domain/remote.php/carddav/addressbooks/**users/**username/contacts_shared_by_adminusername/

**Comment by jobisoft** [comment]
Are you sure, the user has WRITE permissions for the shared folder? The 404 could be wrong and owncloud wanted to say 403 or something like that...

**Comment by jobisoft** [comment]
ok, but it is still the case, that tbsync can read from that URL but cannot write, right?

**Comment by vgencda** [comment]
> ok, but it is still the case, that tbsync can read from that URL but cannot write, right?

Exactly. Just tested if direct write permissions for a specific user make a difference. Unfortunately not.
I had a look at DAVDroid (Android) and CalDav Synchronizer (Outlook/Win10). Those seem to use the Webdav Interface and are able to write to the shared addressbook. 

There seems to be an issue with the carddav URL by the looks of it, like you suggested. Are you in contact with the developers there? The issue should be reproducable with the steps decribed in my first post on any standard installation of owncloud X I think.

**Comment by jobisoft** [comment]
I am not in contact with ownCloud devs, I would have to let you do that.

But can you give a few more details on how they use the WebDAV interface? What are they doing? The Sabre/dav specs are pretty forward on how to do  PUTs.

**Comment by vgencda** [comment]
Hi, sorry for the delay. Unfortunately I have no information on how those applications communicate with the owncloud server. I just noticed that adding new contacts works and looked for differences.

**Comment by vgencda** [comment]
Some more info. TBSync seems to rely on the RewriteRules of owncloud. As pointed out in one of your user reviews the following

RewriteRule ^\.well-known/carddav /remote.php/dav/ [R=301,L]
RewriteRule ^\.well-known/caldav /remote.php/dav/ [R=301,L]

has to be changed to

RewriteRule ^.well-known/carddav /remote.php/carddav/ [R=301,L]
RewriteRule ^.well-known/caldav /remote.php/caldav/ [R=301,L]

for TBSync to work at all with owncloud. This is not optimal as you have to be admin of the owncloud instance in order for TBSync to work. Could you resolve the URLs on another way?

I noticed this also breaks the ability to write to shared adressbooks in CalDav Synchronizer (Outlook/Win10) if you detect the available adressbooks. 

So writing to URLs like owncloud.domain/remote.php/carddav/addressbooks/username/contacts_shared_by_adminusername/ which result from changing the .htaccess is not possible. When I change the URL in the CalDav Synchronizer Profile back to the one the owncloud user interface is giving me (which is owncloud.domain/remote.php/dav/addressbooks/users/username/contacts_shared_by_adminusername/) writing to the shared addressbook works.

So in short, it seems like you can not write to shared addressbooks if you don´t use the WebDav URL that owncloud provides you.

**Comment by jobisoft** [comment]
Sorry for your hassle, but I still think this is a bug in ownCloud. All they need to do is to  announce the correct URL. 

See https://github.com/jobisoft/DAV-4-TbSync/issues/29#issuecomment-435341082

They tell me to use:
/remote.php/carddav/addressbooks/username/contacts_shared_by_adminusername/

But they should tell me to use (this is independent of the rewrite stuff)
/remote.php/dav/addressbooks/users/username/contacts_shared_by_adminusername/

I think you need to discuss that with one of the ownCloud devs. I will assist, if there are follow-up questions.

There is still the possibility I am doing something wrong, but than this only affects owncloud and I have no idea what I should change.

**Comment by vgencda** [comment]
> 
> 
> For the "wrong" rewrite. TbSync does function without it, you just have to give the full URL in the setup.
> 
> Instead of just `my.server.com` use `my.server.com/remote.php/carddav/` and `my.server.com/remote.php/caldav/` (you need to setup two accounts in Tbsync, one for calendar, one for contacts).

Yes, makes sense, will do, thanks :).

**Comment by jobisoft** [comment]
The latest beta allows to specify independent urls for CalDAV and CardDav in manual setup. If you want to try:
tbsync.jobisoft.de

That allows to have calendars and contacts in the same account again.

Can I close this?
